src/test/java/com/sap/sgs/phosphor/fosstars/model/score/AverageCompositeScoreTest.java

```diff
@@ -0,0 +1,176 @@
+    assertTrue(score.description().isEmpty());
+    assertEquals(2, score.subScores().size());
+    assertTrue(score.subScores().contains(new FirstScore()));
+    assertTrue(score.subScores().contains(new SecondScore()));
+  }
+
+  @Test
+  public void equalsAndHashCode() {
+    Score one = new TestAverageCompositeScore();
+    Score two = new TestAverageCompositeScore();
+    assertEquals(one, two);
+    assertEquals(one.hashCode(), two.hashCode());
+    assertNotEquals(null, one);
+    assertEquals(one, one);
+  }
+
+  @Test
+  public void validValue() {
+    Score score = new TestAverageCompositeScore();
+    Value<Double> value = score.value(7.54);
+    assertNotNull(value);
+    assertEquals(7.54, value.get(), PRECISION);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void negativeValue() {
+    new TestAverageCompositeScore().value(-3.0);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void tooBigValue() {
+    new TestAverageCompositeScore().value(42.0);
+  }
+
+  @Test
+  public void allScoresPreCalculated() {
+    final Score score = new TestAverageCompositeScore();
+
+    final double firstValue = 3.0;
+    final double secondValue = 2.0;
+
+    final double firstConfidence = 7.3;
+    final double secondConfidence = 6.1;
+
+    ScoreValue firstPreCalculatedScoreValue = new ScoreValue(new FirstScore());
+    firstPreCalculatedScoreValue.set(firstValue);
+    firstPreCalculatedScoreValue.confidence(firstConfidence);
+
+    ScoreValue secondPreCalculatedScoreValue = new ScoreValue(new SecondScore());
+    secondPreCalculatedScoreValue.set(secondValue);
+    secondPreCalculatedScoreValue.confidence(secondConfidence);
+
+    ScoreValue scoreValue = score.calculate(
+        firstPreCalculatedScoreValue,
+        secondPreCalculatedScoreValue);
+
+    assertNotNull(scoreValue);
+
+    double expectedScore = (firstValue + secondValue) / 2;
+    double expectedConfidence = (firstConfidence + secondConfidence) / 2;
+
+    assertEquals(expectedScore, scoreValue.get(), PRECISION);
+    assertEquals(expectedConfidence, scoreValue.confidence(), PRECISION);
+  }
+
+  @Test
+  public void oneScorePreCalculated() {
+    Score score = new TestAverageCompositeScore();
+
+    final double firstValue = 3.0;
+    final double firstConfidence = 7.2;
+
+    ScoreValue preCalculatedScoreValue = new ScoreValue(new FirstScore());
+    preCalculatedScoreValue.set(firstValue);
+    preCalculatedScoreValue.confidence(firstConfidence);
+
+    ScoreValue scoreValue = score.calculate(preCalculatedScoreValue);
+    assertNotNull(scoreValue);
+
+    double expectedScore = (firstValue + SecondScore.VALUE) / 2;
+    assertEquals(expectedScore, scoreValue.get(), PRECISION);
+
+    double expectedConfidence = (firstConfidence + Confidence.MAX) / 2;
+    assertEquals(expectedConfidence, scoreValue.confidence(), PRECISION);
+  }
+
+  private static class FirstScore extends FeatureBasedScore {
+
```

Comment:
Please remove this line.

Reply:
I would prefer to keep this like since it separates the class definition and fields.

Reply:
ok